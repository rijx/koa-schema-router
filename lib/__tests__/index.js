const createInternalRouter = require("find-my-way");

const createRouter = require("..");

jest.mock("ajv");
jest.mock("find-my-way");
jest.mock("fast-json-stringify");

describe("Router", () => {
  test("define single route without schema", () => {
    const dummyInternalRouter = {
      on: jest.fn()
    };

    createInternalRouter.mockReturnValue(dummyInternalRouter);

    const router = createRouter();

    const dummyRoute = {
      method: "GET",
      url: "/"
    };

    router.define(dummyRoute);

    expect(dummyInternalRouter.on.mock.calls.length).toBe(1);
    expect(dummyInternalRouter.on.mock.calls).toMatchSnapshot();
  });

  test("define single route with schema", () => {
    const dummyInternalRouter = {
      on: jest.fn()
    };

    createInternalRouter.mockReturnValue(dummyInternalRouter);

    const router = createRouter();

    const dummyRoute = {
      method: "GET",
      url: "/",
      schema: {
        request: {
          query: { type: "object" },
          params: { type: "object" },
          body: { type: "object" }
        },
        response: {
          200: { type: "object" }
        }
      }
    };

    router.define(dummyRoute);

    expect(dummyInternalRouter.on.mock.calls.length).toBe(1);
    expect(dummyInternalRouter.on.mock.calls[0]).toMatchSnapshot();
  });
});
