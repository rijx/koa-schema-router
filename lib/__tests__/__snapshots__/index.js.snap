// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router define single route with schema 1`] = `
Array [
  "GET",
  "/",
  undefined,
  Object {
    "schema": Object {
      "request": Object {
        "body": Object {
          "type": "object",
        },
        "params": Object {
          "type": "object",
        },
        "query": Object {
          "type": "object",
        },
      },
      "response": Object {
        "200": Object {
          "type": "object",
        },
      },
    },
    "serializers": Map {
      "200" => undefined,
    },
    "validators": Object {
      "body": undefined,
      "params": undefined,
      "query": undefined,
    },
  },
]
`;

exports[`Router define single route without schema 1`] = `
Array [
  Array [
    "GET",
    "/",
    undefined,
    Object {
      "schema": undefined,
      "serializers": Map {},
      "validators": Object {
        "body": undefined,
        "params": undefined,
        "query": undefined,
      },
    },
  ],
]
`;
